### PR TITLE
Define Element interface extensions

### PR DESCRIPTION
**Pull request by jeisinger**
Also change the spelling of the IDL attribute to referrerPolicy

@annevk @estark37 @mikewest @hsivonen -- any comments?

### CODE REVIEW

File: index.src.html [hunk 1]
```diff
@@ -380,7 +382,7 @@ <h2 id="referrer-policy-delivery">Referrer Policy Delivery</h2>
       Via a <{meta}> element with a <{meta/name}> of <code>referrer</code>.
     </li>
     <li>
-      Via a <code>referrerpolicy</code> attribute on an <{a}>, <{area}>, <{img}>
+      Via a <code>referrerPolicy</code> attribute on an <{a}>, <{area}>, <{img}>
```
Comment (annevk) [inline index.src.html]: This should remain lowercase.

File: index.src.html [hunk 2]
```diff
@@ -516,18 +518,18 @@ <h3 id="referrer-policy-delivery-meta">Delivery via <{meta}></h3>
   </dl>
 
   <h3 id="referrer-policy-delivery-referrer-attribute">Delivery
-  via a <code>referrerpolicy</code> attribute</h3>
+  via a <code>referrerPolicy</code> attribute</h3>
```
Comment (annevk) [inline index.src.html]: This too.

File: index.src.html [hunk 3]
```diff
@@ -516,18 +518,18 @@ <h3 id="referrer-policy-delivery-meta">Delivery via <{meta}></h3>
...
 
-  A referrer policy may be set when a <code>referrerpolicy</code> attribute is
+  A referrer policy may be set when a <code>referrerPolicy</code> attribute is
```
Comment (annevk) [inline index.src.html]: This too.

File: index.src.html [hunk 4]
```diff
@@ -516,18 +518,18 @@ <h3 id="referrer-policy-delivery-meta">Delivery via <{meta}></h3>
...
   added to an <code><a element>a</a></code>, <code><a element>area</a></code>,
   <code><a element>img</a></code> or <code><a element>iframe</a></code>
   element, for example:
 
   <pre class="example">
-    &lt;a href="http://example.com" referrerpolicy="origin"&gt;
+    &lt;a href="http://example.com" referrerPolicy="origin"&gt;
```
Comment (annevk) [inline index.src.html]: This too.

File: index.src.html [hunk 5]
```diff
@@ -516,18 +518,18 @@ <h3 id="referrer-policy-delivery-meta">Delivery via <{meta}></h3>
...
   </pre>
 
-  The following values for the <code>referrerpolicy</code> attribute are valid,
+  The following values for the <code>referrerPolicy</code> attribute are valid,
```
Comment (annevk) [inline index.src.html]: This too.

File: index.src.html [hunk 6]
```diff
@@ -543,13 +545,13 @@ <h3 id="referrer-policy-delivery-referrer-attribute">Delivery
     <dd><a><code>Unsafe URL</code></a></dd>
   </dl>
 
-  A policy delivered via a <code>referrerpolicy</code> attribute on an element
+  A policy delivered via a <code>referrerPolicy</code> attribute on an element
```
Comment (annevk) [inline index.src.html]: This too.

File: index.src.html [hunk 7]
```diff
@@ -543,13 +545,13 @@ <h3 id="referrer-policy-delivery-referrer-attribute">Delivery
...
   takes precedence over the policy defined for the whole document via CSP or
   a <a element>meta</a> element unless the attribute value is invalid.
 
   NOTE: If an <code><a element>a</a></code>
   or <code><a element>area</a></code> element includes both
-  a <code>referrerpolicy</code> attribute as well as
+  a <code>referrerPolicy</code> attribute as well as
```
Comment (annevk) [inline index.src.html]: This too.

File: index.src.html [hunk 8]
```diff
@@ -606,6 +608,50 @@ <h4 id="referrer-policy-delivery-implicit-workers">
...
+
+  <h3 id="html-anchor-element">HTMLAnchorElement</h3>
+  <pre class="idl">
+    partial interface HTMLAnchorElement {
+      attribute DOMString referrerPolicy;
+    };
+  </pre>
+  <h4 id="html-anchor-element-attributes">Attributes</h4>
+  The <code>referrerPolicy</code> IDL attribute must <a>reflect</a> the
+  <a>settings object</a>'s referrer policy, <a>limited to only known values</a>.
```
Comment (annevk) [inline index.src.html]: This doesn't make sense. The IDL attribute must reflect the `referrerpolicy` content attribute.

File: index.src.html [hunk 9]
```diff
@@ -606,6 +608,50 @@ <h4 id="referrer-policy-delivery-implicit-workers">
...
+
+  <h3 id="html-area-element">HTMLAreaElement</h3>
+  <pre class="idl">
+    partial interface HTMLAreaElement {
+      attribute DOMString referrerPolicy;
+    };
+  </pre>
+  <h4 id="html-area-element-attributes">Attributes</h4>
+  The <code>referrerPolicy</code> IDL attribute must <a>reflect</a> the
+  <a>settings object</a>'s referrer policy, <a>limited to only known values</a>.
```
Comment (annevk) [inline index.src.html]: Same here.

File: index.src.html [hunk 10]
```diff
@@ -606,6 +608,50 @@ <h4 id="referrer-policy-delivery-implicit-workers">
...
+
+  <h3 id="html-image-element">HTMLImageElement</h3>
+  <pre class="idl">
+    partial interface HTMLImageElement {
+      attribute DOMString referrerPolicy;
+    };
+  </pre>
+  <h4 id="html-image-element-attributes">Attributes</h4>
+  The <code>referrerPolicy</code> IDL attribute must <a>reflect</a> the
+  <a>settings object</a>'s referrer policy, <a>limited to only known values</a>.
```
Comment (annevk) [inline index.src.html]: Same here.

File: index.src.html [hunk 11]
```diff
@@ -606,6 +608,50 @@ <h4 id="referrer-policy-delivery-implicit-workers">
...
+
+  <h3 id="html-iframe-element">HTMLIFrameElement</h3>
+  <pre class="idl">
+    partial interface HTMLIFrameElement {
+      attribute DOMString referrerPolicy;
+    };
+  </pre>
+  <h4 id="html-iframe-element-attributes">Attributes</h4>
+  The <code>referrerPolicy</code> IDL attribute must <a>reflect</a> the
+  <a>settings object</a>'s referrer policy, <a>limited to only known values</a>.
```
Comment (annevk) [inline index.src.html]: Same here.

**Comment by jeisinger** [comment]
ptal

**Pull request merged**